mobile/src/main/java/technology/mainthread/apps/grandmaps/service/GrandMapsArtSourceService.java

```diff
@@ -163,13 +155,22 @@ public void displayRefreshInfo(SharedPreferences sharedPreferences) {
         displayToast("Next update time: " + dateFormat.format(new Date(nextUpdateTimeMillis)));
     }
 
-    private Artwork convertResponseToArtwork(GrandMapsResponse responseBody) {
+    private ImageResponse getRandomImage(List<ImageResponse> images) {
+        if (images ==  null || images.isEmpty()) {
+            return null;
+        }
+
+        int size = images.size();
+        int nextIndex = size > 1 ? random.nextInt(size - 1) : 0;
```

Comment:
http://stackoverflow.com/a/363692/88588, could make this magic a little more clear.